src/state/psarstake/hederaHooks.ts

```diff
@@ -196,6 +208,191 @@ export function useDerivativeHederaSarStake(positionId?: BigNumber) {
   );
 }
 
+/**
+ * This hook return the rent value of a position
+ * @param positionId The id of position
+ * Returns rent value in tiny bars
+ */
+function useHederaRent(positionId: string | undefined) {
+  const sarStakingContract = useSarStakingContract();
+  const blockTimestamp = useGetBlockTimestamp();
+
+  const positionState = useSingleCallResult(
+    positionId ? sarStakingContract : undefined,
+    'positions',
+    positionId ? [positionId] : [],
+  );
+
+  const { data: exchangeRate, isLoading: isLoadingRate } = useQuery(
+    'get-hedera-exchange-rate',
+    async () => {
+      const rate = await hederaFn.getExchangeRate();
+      return rate;
+    },
+    {
+      cacheTime: 60 * 1000, // 1 minute
+    },
+  );
+
+  /*
+    rentTime = block.timestamp - position.lastUpdate;
+    rentAmount = rentTime * tinyCentsToTinyBars(500_000_000) / (90 days/s);
+  */
+
+  return useMemo(() => {
+    const isLoading = isLoadingRate || positionState.loading;
+    if (!positionId || !blockTimestamp || isLoading || !exchangeRate || positionState.error || !positionState.valid) {
+      return undefined;
+    }
+
+    try {
+      const tinyBars = hederaFn.tinyCentsToTinyBars('500000000', exchangeRate.current_rate);
+      const lastUpdate = positionState.result?.lastUpdate;
+      const rentTime = Number(blockTimestamp) - lastUpdate;
+      const days = 90 * 24 * 60 * 60;
+      const rentAmount = JSBI.divide(
+        JSBI.multiply(JSBI.BigInt(rentTime.toString()), JSBI.BigInt(tinyBars)),
+        JSBI.BigInt(days.toString()),
+      );
+      return new Fraction('12', '10').multiply(rentAmount).toFixed(0);
```

Comment:
1. can you please add some comment on what are magic number `500000000`, `12`, `10` stands for ?
2. why are we using 2 different kind of technique to fine rent ? one is coingecko technique and other is exchange rate technique. I am more interested to know it better.

Reply:
1 - This numbers was provided by Efe
2 - I made the coingecko before the chain exchangerate, I will change this